cli/flox-rust-sdk/src/models/user_state.rs

```diff
@@ -0,0 +1,93 @@
+use std::path::{Path, PathBuf};
+
+use flox_core::{serialize_atomically, traceable_path, SerializeError};
+use fslock::LockFile;
+use serde::{Deserialize, Serialize};
+use tracing::debug;
+
+use crate::flox::Flox;
+
+pub const USER_STATE_FILENAME: &str = "user_state.json";
+
+#[derive(Debug, thiserror::Error)]
+pub enum UserStateError {
+    #[error("couldn't acquire user state file lock")]
+    AcquireLock(#[source] fslock::Error),
+    #[error("couldn't read user state file")]
+    ReadFile(#[source] std::io::Error),
+    #[error("couldn't parse user state file")]
+    Parse(#[source] serde_json::Error),
+    #[error("failed to write user state file")]
+    WriteFile(#[source] SerializeError),
+}
+
+#[derive(Debug, Clone, Serialize, Deserialize, Default)]
+pub struct UserState {
+    pub confirmed_create_default_env: Option<bool>,
+}
+
+// TODO: These functions are very close to their counterparts in `env_registry.rs`.
+//       The main differences are error types. We could share a common set of functionality
+//       by creating a trait that uses associated types/constants to identify the error types
+//       that are used at the different steps, then provide default implementations for the
+//       operations since they're essentially identical.
+
+/// Returns the path to the user's state file.
+pub fn user_state_path(flox: &Flox) -> PathBuf {
+    flox.data_dir.join(USER_STATE_FILENAME)
```

Comment:
nit: we store update-notification.json in cache so I'd probably do the same for this

Reply:
Fixed